Code/Source/Entity/EntityManager.cpp

```diff
@@ -87,4 +97,31 @@ namespace RGL
         }
         m_isPoseUpdateNeeded = false;
     }
+
+    int32_t EntityManager::getRGLId() {
+        uint8_t classId = 0;
+        if (!m_classTags) {
+            return 1;
+        }
+        for (const auto &tagNameAndClassId: *m_classTags) {
+            LmbrCentral::Tag tag(tagNameAndClassId.first);
+            bool hasTag = false;
+            LmbrCentral::TagComponentRequestBus::EventResult(hasTag, m_entityId,
+                                                             &LmbrCentral::TagComponentRequests::HasTag, tag);
+            if (hasTag) {
+                if (classId != 0) {
+                    AZ_Warning("EntityManager", false,
+                               "Entity with ID: %s has more than one class tag. Assigning tag: %s",
+                               m_entityId.ToString().c_str(), tagNameAndClassId.first.c_str());
+                }
+                classId = tagNameAndClassId.second;
+            }
+        }
+        if (classId == 0) {
+            AZ_Warning("EntityManager", false, "Entity with ID: %s has no class tag. Assigning default class ID: 0",
+                       m_entityId.ToString().c_str());
+        }
```

Comment:
Line 104 above - I feel like this case here and the one above both should return default id, that is either 0 or 1 or whatever we will agree one. Just be sure it is consistent.

Reply:
yaaa, my bad in general that thing on line 104 was still in class 0, because 1 is packed value, i changed this a bit and now it should make more sense and be more self consistent

Reply:
Updated version is nice!